pennylane/operation.py

```diff
@@ -483,6 +477,16 @@ def __init__(self, *params, wires=None, do_queue=True, id=None):
         if wires is None:
             raise ValueError("Must specify the wires that {} acts on".format(self.name))
 
+        self._num_params = len(params)
+        # Check if the expected number of parameters coincides with the one received.
+        # This is always true for the default `Operator.num_params` property, but
+        # subclasses may overwrite it to define a fixed expected value.
+        if len(params) != self.num_params:
```

Comment:
wait, won't this always be true? since `len(params)` is setting the `num_params`, which it then checks?

Reply:
Ah I hoped the comment would clarify this! 

The operator subclasses where the number of expected params is known overwrite the property `num_params`. So it is always true for the default property, but 98% of the subclasses overwrite it, in which case the check is meaningful.

Reply:
got it!